modules/core/test/test_io.cpp

```diff
@@ -756,93 +779,10 @@ static void test_filestorage_basic(int write_flags, const char* suffix_name, boo
-        ASSERT_EQ(_2d_in_u64.cols   , _2d_out_u64.cols);
-        ASSERT_EQ(_2d_in_u64.dims   , _2d_out_u64.dims);
-        ASSERT_EQ(_2d_in_u64.depth(), _2d_out_u64.depth());
-
-        errors = 0;
-        for(int i = 0; i < _2d_out_u8.rows; ++i)
-        {
-            for (int j = 0; j < _2d_out_u8.cols; ++j)
-            {
-                if (_2d_in_u8.at<cv::Vec3b>(i, j) != _2d_out_u8.at<cv::Vec3b>(i, j)) {
-                    EXPECT_EQ(_2d_in_u8.at<cv::Vec3b>(i, j), _2d_out_u8.at<cv::Vec3b>(i, j));
-                    printf("i = %d, j = %d\n", i, j);
-                    if (++errors >= 3)
-                    {
-                        i = _2d_out_u8.rows;
-                        break;
-                    }
-                }
-            }
-        }
-
-        errors = 0;
-        for(int i = 0; i < _2d_out_u32.rows; ++i)
-        {
-            for (int j = 0; j < _2d_out_u32.cols; ++j)
-            {
-                if (_2d_in_u32.at<cv::Vec<uint, 3>>(i, j) != _2d_out_u32.at<cv::Vec<uint, 3>>(i, j)) {
-                    EXPECT_EQ((_2d_in_u32.at<cv::Vec<uint, 3>>(i, j)), (_2d_out_u32.at<cv::Vec<uint, 3>>(i, j)));
-                    printf("i = %d, j = %d\n", i, j);
-                    if (++errors >= 3)
-                    {
-                        i = _2d_out_u32.rows;
-                        break;
-                    }
-                }
-            }
-        }
-
-        errors = 0;
-        for(int i = 0; i < _2d_out_i64.rows; ++i)
-        {
-            for (int j = 0; j < _2d_out_i64.cols; ++j)
-            {
-                if (_2d_in_i64.at<cv::Vec3l>(i, j) != _2d_out_i64.at<cv::Vec3l>(i, j)) {
-                    EXPECT_EQ(_2d_in_i64.at<cv::Vec3l>(i, j), _2d_out_i64.at<cv::Vec3l>(i, j));
-                    printf("i = %d, j = %d\n", i, j);
-                    if (++errors >= 3)
-                    {
-                        i = _2d_out_i64.rows;
-                        break;
-                    }
-                }
-            }
-        }
-
-        errors = 0;
-        for(int i = 0; i < _2d_out_u64.rows; ++i)
-        {
-            for (int j = 0; j < _2d_out_u64.cols; ++j)
-            {
-                if (_2d_in_u64.at<cv::Vec<uint64_t, 3>>(i, j) != _2d_out_u64.at<cv::Vec<uint64_t, 3>>(i, j)) {
-                    EXPECT_EQ((_2d_in_u64.at<cv::Vec<uint64_t, 3>>(i, j)), (_2d_out_u64.at<cv::Vec<uint64_t, 3>>(i, j)));
-                    printf("i = %d, j = %d\n", i, j);
-                    if (++errors >= 3)
-                    {
-                        i = _2d_out_u64.rows;
-                        break;
-                    }
-                }
-            }
-        }
+        compare_normal_2d_mats<Vec3b>(_2d_in_u8, _2d_out_u8);
+        compare_normal_2d_mats<Vec<uint, 3>>(_2d_in_u32, _2d_out_u32);
+        compare_normal_2d_mats<Vec3l>(_2d_in_i64, _2d_out_i64);
+        compare_normal_2d_mats<Vec<uint64_t, 3>>(_2d_in_u64, _2d_out_u64);
```

Comment:
`EXPECT_MAT_NEAR(_2d_in_u8, _2d_out_u8, 0)` or `EXPECT_MAT_N_DIFF(_2d_in_u8, _2d_out_u8, 0)` or `EXPECT_PRED_FORMAT2(cvtest::MatComparator(0, 0), _2d_in_u8, _2d_out_u8);` should be enough here. The in and out matrices should be bit exact, right?